src/qt/diagnosticsdialog.cpp

```diff
@@ -266,33 +263,55 @@ void DiagnosticsDialog::on_testBtn_clicked() {
 
 }
 
-void DiagnosticsDialog::clkFinished() {
-    time_t ntpTime(0);
-
-    while(udpSocket->pendingDatagramSize() != -1) {
-        char recvBuffer[udpSocket->pendingDatagramSize()];
-        udpSocket->readDatagram(&recvBuffer[0], 48, nullptr, nullptr);
-        if(sizeof(recvBuffer) == 48){
-            int count= 40;
-            unsigned long DateTimeIn= uchar(recvBuffer[count])+ (uchar(recvBuffer[count+ 1]) << 8)+ (uchar(recvBuffer[count+ 2]) << 16)+ (uchar(recvBuffer[count+ 3]) << 24);
-            ntpTime = ntohl((time_t)DateTimeIn);
-            ntpTime -= 2208988800U;
-        }
-    }
+void DiagnosticsDialog::clkStateChanged(QAbstractSocket::SocketState state)
+{
+    if (state == QAbstractSocket::ConnectedState)
+    {
+        connect(udpSocket, SIGNAL(readyRead()), this, SLOT(clkFinished()));
 
-    udpSocket->close();
+        char NTPMessage[48] = {010, 0, 0, 0 ,0, 0, 0, 0, 0};
```

Comment:
I don't think 010 is correct. That's 10 octal or 8 decimal. I think we can leave it at 0:

`char NTPMessage[48] = {};`

or, if the compiler complains:

`char NTPMessage[48] = { 0 };`

Reply:
the old code was 
QByteArray sendBuffer = QByteArray(47, 0);
 -    sendBuffer.insert(0, 35);

i didnt understand that but the new code i used to get this to work properly and referenced sent that NTPMessage. idk the NTP protocol. i'm assuming the old code send 35 0's?

Reply:
if so a 0 should be fine i'l test it

Reply:
apon setting as you asked the Verify Clock is stuck on testing. sp that NTPMessage must be correct as i had it

Reply:
Oh, try it with `0x1b` for the first byte.

Reply:
trying char NTPMessage[48] = {0x1b, 0, 0, 0 ,0, 0, 0, 0, 0};

Reply:
works with char NTPMessage[48] = {0x1b, 0, 0, 0 ,0, 0, 0, 0, 0};  
u satisfied with that boss man

Reply:
I am, that seems to set the correct bits.